beast/tools/remove_filters.py

```diff
@@ -93,41 +105,96 @@ def remove_filters_from_files(
+                        if csfilter in beast_filt_short:
+
+                            # if it's the same instrument, delete it
+                            if beast_filt[beast_filt_short.index(csfilter)] == cfilter:
+                                rindxs.append(filters.index(cfilter))
+                                for grid_col in g0.grid.colnames:
+                                    if cfilter in grid_col:
+                                        rgridcols.append(grid_col)
+
+                            # if it's not the same instrument, keep it
+                            else:
+                                continue
+
+                        # if the current filter isn't in the BEAST ref list, delete it
+                        else:
+                            rindxs.append(filters.index(cfilter))
+                            for grid_col in g0.grid.colnames:
+                                if cfilter in grid_col:
+                                    rgridcols.append(grid_col)
+
+                    # if there isn't a list of BEAST refs, delete it
+                    else:
+                        rindxs.append(filters.index(cfilter))
+                        for grid_col in g0.grid.colnames:
+                            if cfilter in grid_col:
+                                rgridcols.append(grid_col)
+
+
+            # --------------------------
             # if the removed filters are determined from the catalog file
             if rm_filters is None:
+
+                # if the current filter is present in the catalog filters
                 if csfilter in cat_filters:
-                    nlamb.append(clamb)
-                    nfilters.append(cfilter)
+
+                    # if there's a list of BEAST instrument+filter references
+                    if beast_filt is not None:
+
+                        # if the current filter is in the list of BEAST references
+                        if csfilter in beast_filt_short:
+
+                            # if it's the same instrument, keep it
+                            if beast_filt[beast_filt_short.index(csfilter)] == cfilter:
+                                continue
+
+                            # if it's not the same instrument, delete it
+                            else:
+                                rindxs.append(filters.index(cfilter))
+                                for grid_col in g0.grid.colnames:
+                                    if cfilter in grid_col:
+                                        rgridcols.append(grid_col)
+
+                        # if the current filter isn't in the BEAST ref list, keep it
+                        else:
+                            continue
+
+                    # if there isn't a list of BEAST refs, keep it
+                    else:
+                        continue
+
+                # if the current filter isn't in the catalog filters, delete it
```

Comment:
Why are the continue statements needed?  It seems that the if statements could be reconfigured to provide the same results since everything(?) is already embedded in if statements.